src/expr/src/relation/func.rs

```diff
@@ -978,6 +978,59 @@ fn generate_series_ts(
     Ok(tsri.map(move |i| (Row::pack_slice(&[conv(i)]), 1)))
 }
 
+fn generate_subscripts_array(
+    a: Datum,
+    dim: i32,
+) -> Result<Box<dyn Iterator<Item = (Row, Diff)>>, EvalError> {
```

Comment:
It'd be good to write this without boxing. It shouldn't be too hard! You just need to funnel all paths into a single call to `generate_series`, where the empty case results in `generate_series(1, 0, 1)` to produce no rows.

Reply:
When reading the code, `generate_series(1, 0, 1)` is really awkward. Unless you have strong opinions, I would like to prioritize readability over slight performance gains. This code is not performance-critical.